### PR TITLE
[E2E refactor - 3] re-introduce generated clientsets to prepare for simpler e2e tests

### PR DESCRIPTION
**Pull request by pintohutch**
I decided to break out the kind E2E test refactor PR https://github.com/GoogleCloudPlatform/prometheus-engine/pull/738 into smaller, digestible PRs for reviewing.

Note: because of the nature of the change, presubmits (i.e. Github Actions) may fail until the final PR is merged.

This is the third one, where we reintroduced generating k8s clientsets of our CRD types. This sets us up to ensure e2e tests don't require controller-runtime, which can be beneficial when importing the library elsewhere. This is an effective revert of c5eeee3209b63920ba888f2025f127ac4202564c.

### CODE REVIEW

**Review by bwplotka — APPROVED** [review]
👍🏽 LGTM

**Comment by pintohutch** [comment]
Thanks! Waiting on @TheSpiritXIII's review on #744 and once that looks good, i'll merge this as well 👍

**Pull request merged**
